CMakeLists.txt

```diff
@@ -1604,6 +1604,12 @@ endif()
 hpx_libraries(${HWLOC_LIBRARIES})
 include_directories(${HWLOC_INCLUDE_DIR})
 
+################################################################################
+# Set search paths for generated and libs headers
+################################################################################
+include_directories("${CMAKE_BINARY_DIR}/include/generated")
+include_directories("${CMAKE_BINARY_DIR}/include/libs")
```

Comment:
The main reason I am reluctant to have those here is that "hidden" dependencies are masked that way. We won't notice if we forgot to add some dependencies while developing our new modules.
This is exactly the hack which renders the whole effort to be ignored. Without those lines, we have to invest something now, rather than invest heavily again once we decide to remove them, because we think everything is done.

Reply:
You're absolutely right. I'm not even sure the libs ones are needed right now. I'll try removing them. The generated ones I'm not sure about. I'd like to see them gone as well from the global includes but not sure how. APEX finds e.g. the version header only because of this right now. Suggestions on how to avoid that?